app/models/updates_production_credits.rb

```diff
@@ -0,0 +1,54 @@
+class UpdatesProductionCredits
+  def self.update(generic_file)
+    new(generic_file).update
+  end
+
+  def initialize(generic_file)
+    @generic_file = generic_file
+  end
+
+  def update
+    update_productions
+    update_venues
+    update_works
+  end
+
+  private
+
+  attr_reader :generic_file
+
+  def update_productions
+    generic_file.production_names = productions.map(&:production_name).compact
```

Comment:
It would have to be `all_production_names`, since it's only the names.  I'm not sure it's worth pulling out - it's only used here.